site/source/docs/porting/connecting_cpp_and_javascript/Interacting-with-code.rst

```diff
@@ -342,6 +342,168 @@ If you add it to your own file, you should write something like
 first, so this add ``my_js`` onto ``LibraryManager.library``, the global
 object where all JavaScript library code should be.
 
+JavaScript Limits in library files
+----------------------------------
+
+If you're not familar with JavaScript, like you're a C/C++ programmer
+and just using emscripten these issues probably won't come up but
+if you're an experienced JavaScript programmer you need to be aware
+many common JavaScript practices can not be used in emscripten
+library files.
+
+To save space, by default, emscripten only includes library properties
+referenced from C/C++. It does this by calling ``toString`` on each 
+used property. Conceptually you can consider it like this
+
+.. code-block:: javascript
+
+   usedLibraryFunctions.forEach(function(name) {
+      var func = LibraryManager.library[name];
+      emitCode("function _" + name + func.toString().replace("function", ""));
+   });
+   
+It's actually not that simple as it will emit JavaScript objects
+but any functions they reference must be expressible by calling `toString`.
+   
+The implications of that are as follows
+
+-  You can not use ES6 object shorthand function syntax.
+-  You can not reference/bind other functions directly.
+-  You can not use a closure directly.
+-  You can not use ES6 arrow syntax.
+-  You can not use other function transformations like currying.
+
+.. code-block:: javascript
+
+   mergeInto(LibraryManager.library, {
+     // bad: ES6 object shorthand function syntax not allowwed
+     bad_01() {
+       alert('hi');
+     },
+     
+     // bad: Can not bind/reference other functions directly.
+     bad_02: document.querySelector.bind(document),
+     
+     // bad: You can not use a closure directly
+     bad_03: (function() {
+       var callCount = 0;
+       return function() {
+         console.log("times called: ", ++callCount);
+       };
+     }()),
+     
+     // bad: You can not use ES6 arrow syntax
+     bad_04: () => { console.log("hi!"); },
+     
+     // bad: You can not curry/transform
+     bad_05: curry(scrollTo, 0),  
+   });
+   
+Workarounds to the above issues
```

Comment:
I would remove this line, and start the next line with "To avoid these issues,", since pre-js is the standard way to add arbitrary JS, it's not a workaround in the common sense of "hack to make things work" ;)